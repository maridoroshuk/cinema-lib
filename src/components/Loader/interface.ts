export interface ILoaderProps {
  height: number;
  width: number;
  color: string;
}
