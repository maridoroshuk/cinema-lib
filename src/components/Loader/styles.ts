import styled from 'styled-components';

export const LoaderWrapper = styled.div`
  width: 100%;
  height: 100%;
  position: absolute;
  top: 50%; right: 50%;
  transform: translate(50%,-50%);
`;
