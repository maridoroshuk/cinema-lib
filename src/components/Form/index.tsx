import React from 'react';
import { Formik } from 'formik';

import { IFormProps } from './interface';
import { StyledForm } from './styles';

export const Form = <T extends {}>({
  children,
  initialValues,
  validate,
  onFormSubmit,
}: IFormProps<T>) => {
  return (
    <Formik
      initialValues={initialValues}
      validate={validate}
      onSubmit={onFormSubmit}
    >
      <StyledForm>{children}</StyledForm>
    </Formik>
  );
};
