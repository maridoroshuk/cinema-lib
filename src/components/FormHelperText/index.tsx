import React from 'react';

import { IFormHelperTextProps } from './interface';
import { Wrapper } from './styles';

export const FormHelperText = ({ message }: IFormHelperTextProps) => {
  return <Wrapper>{message}</Wrapper>;
};
