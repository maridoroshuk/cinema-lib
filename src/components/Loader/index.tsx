import React from 'react';
import ReactLoading from 'react-loading';
import { ILoaderProps } from './interface';
import { LoaderWrapper } from './styles';

export function Loader({ height, width, color }: ILoaderProps) {
  return (
    <LoaderWrapper>
      <ReactLoading type="spokes" color={color} height={height} width={width} />
    </LoaderWrapper>
  );
}
