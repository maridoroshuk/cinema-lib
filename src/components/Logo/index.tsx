import React from 'react';

import LogoImage from '@/assets/Logo.png';
import { getLogoSize } from '@/helpers/getLogoSize';

import { LogoStyled } from './styles';
import { ILogo } from './interface';

export const Logo = ({ size }: ILogo) => {
  const width = getLogoSize(size);
  return (
    <LogoStyled src={LogoImage} width={width} alt="company logo (Modsen)" />
  );
};
