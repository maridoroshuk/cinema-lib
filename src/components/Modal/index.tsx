import React from 'react';

import { IModal } from './interface';
import { Wrapper } from './styles';

export const Modal = ({ children }: IModal) => {
  return <Wrapper>{children}</Wrapper>;
};
