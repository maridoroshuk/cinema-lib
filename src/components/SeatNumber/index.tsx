import React from 'react';

import { ISeatProps } from '@/components/SeatNumber/interface';
import { StyledSeat } from '@/components/SeatNumber/styles';

export const SeatNumber = ({
  children,
  backgroundColor,
  index,
}: Partial<ISeatProps>) => {
  return (
    <StyledSeat backgroundColor={backgroundColor} seatIndex={index}>
      {children}
    </StyledSeat>
  );
};
