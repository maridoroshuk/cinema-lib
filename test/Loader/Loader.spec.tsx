import React from 'react';
import { render } from '@testing-library/react';

import { Loader } from '@/components/Loader';

import 'jest-styled-components';

import { setupTestProvider } from '../setupTests';

describe('Loader', () => {
  it('renders', () => {
    const { container } = render(
      setupTestProvider(<Loader color="black" width={250} height={250} />),
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
