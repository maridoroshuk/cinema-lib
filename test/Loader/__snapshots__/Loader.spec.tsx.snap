// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders 1`] = `
.c0 {
  width: 100%;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  class="c0"
>
  <div
    style="fill: black; height: 250px; width: 250px;"
  >
    <svg
      id="loading"
      viewBox="0 0 32 32"
      xmlns="http://www.w3.org/2000/svg"
    >
      
  
      <path
        d="M14 0 H18 V8 H14 z"
        opacity=".1"
        transform="rotate(0 16 16)"
      >
        
    
        <animate
          attributeName="opacity"
          begin="0"
          dur="1s"
          from="1"
          repeatCount="indefinite"
          to=".1"
        />
        
  
      </path>
      
  
      <path
        d="M14 0 H18 V8 H14 z"
        opacity=".1"
        transform="rotate(45 16 16)"
      >
        
    
        <animate
          attributeName="opacity"
          begin="0.125s"
          dur="1s"
          from="1"
          repeatCount="indefinite"
          to=".1"
        />
        
  
      </path>
      
  
      <path
        d="M14 0 H18 V8 H14 z"
        opacity=".1"
        transform="rotate(90 16 16)"
      >
        
    
        <animate
          attributeName="opacity"
          begin="0.25s"
          dur="1s"
          from="1"
          repeatCount="indefinite"
          to=".1"
        />
        
  
      </path>
      
  
      <path
        d="M14 0 H18 V8 H14 z"
        opacity=".1"
        transform="rotate(135 16 16)"
      >
        
    
        <animate
          attributeName="opacity"
          begin="0.375s"
          dur="1s"
          from="1"
          repeatCount="indefinite"
          to=".1"
        />
        
  
      </path>
      
  
      <path
        d="M14 0 H18 V8 H14 z"
        opacity=".1"
        transform="rotate(180 16 16)"
      >
        
    
        <animate
          attributeName="opacity"
          begin="0.5s"
          dur="1s"
          from="1"
          repeatCount="indefinite"
          to=".1"
        />
        
  
      </path>
      
  
      <path
        d="M14 0 H18 V8 H14 z"
        opacity=".1"
        transform="rotate(225 16 16)"
      >
        
    
        <animate
          attributeName="opacity"
          begin="0.675s"
          dur="1s"
          from="1"
          repeatCount="indefinite"
          to=".1"
        />
        
  
      </path>
      
  
      <path
        d="M14 0 H18 V8 H14 z"
        opacity=".1"
        transform="rotate(270 16 16)"
      >
        
    
        <animate
          attributeName="opacity"
          begin="0.75s"
          dur="1s"
          from="1"
          repeatCount="indefinite"
          to=".1"
        />
        
  
      </path>
      
  
      <path
        d="M14 0 H18 V8 H14 z"
        opacity=".1"
        transform="rotate(315 16 16)"
      >
        
    
        <animate
          attributeName="opacity"
          begin="0.875s"
          dur="1s"
          from="1"
          repeatCount="indefinite"
          to=".1"
        />
        
  
      </path>
      

    </svg>
    

  </div>
</div>
`;
